First iteration of governance self assessment guide (#520)
cncf/tag-contributor-strategy: website/content/maintainers/governance/governance-self-assessment.md

---
title: "Governance Self Assessment"
date: 2023-09-15
weight: 40
---

Governance reviews contribute to the health and sustainibility of the CNCF projects. By providing guidance on
effective governance practices, TAG Contributor Strategy aims to ensure that projects operate efficiently,
encourage diverse participation, and uphold the values of the CNCF. The governance review process is designed to be
constructive and supportive, aiming to assist projects in refining their governance models and addressing
any challenges they may face.

Projects may ask TAG Contributor Strategy for assistance in resolving any issues uncovered by the review. The TAG is
available via our [slack channel](https://cloud-native.slack.com/archives/CT6CWS1JN),
[email](https://lists.cncf.io/g/cncf-tag-contributor-strategy),
[GitHub](https://github.com/cncf/tag-contributor-strategy), or by joining our weekly meetings
(listed on the [CNCF public calendar](https://www.cncf.io/calendar/)).

TODO: LATER: How to request a TAG CS governance review?

## General Instructions

> Instructions:

Reviewer: I think that before we ask them to jump into the assessment, it might be easier if they start by updating their governance to use one of our templates. Maybe something like ...

If your project governance was not already based on one of the CNCF governance templates, we recommend that you start by looking at those templates to see if one would be suitable for your project. If so, you can update your governance using one of our templates as pre-work before starting this self-assessment. It isn't a requirement to use one of our templates, but the templates do have most of the elements that we expect to see in project governance documents, so incorporating any missing elements into your governance is likely to make the assessment process easier.

Author: Added the text you've suggested

- Copy and paste the template in [Self Assessment Template](governance-self-assessment-template.md) into your favorite editor.
- Fill in the template with the information about your project. The `TODO` placeholders in the template are there to help you.
- Pick the latest state of the governance document and use the same point-in-time reference (e.g. commit) for your assessment. If governance consists of multiple documents, make sure you use the same commit.
    - TODO: what about websites?
- Make sure all links in your report are permanent links (e.g. with commit hashes).
- To ensure consistency, use the same commits as links for multiple files within a single repository.
- It is recommended to start filling this template from the "Review" section. Once you fill everything there, go back to "Summary and Assessment" to summarize the content you wrote.
    - TODO: do we even need a "Summary and Assessment" section for self-assessments?

## Template details

### Project Information

Fill in the blanks in this section.

##### Executing the Assessment

> * Write a brief description that details the timebox the assessment occurred and the individuals involved in the assessment.

Reviewer: ```suggestion
* Write a brief description that details the timeframe when the assessment occurred and the individuals involved in the assessment.
```
It seemed awkwardly worded to me?

> * The description should contain the reason for the assessment.

Reviewer: ```suggestion
* The description should contain the reason for the assessment. This is usually when the project is applying to change levels, but could occur at other points.
```

> **_NOTE:_** Make sure you use a snapshot of the governance documents for your assessment and note the commit hash of the snapshot here as a link.

#### Review

##### Governance Description

* Write a narrative describing
    * governance type of the project

>     * number of substantial contributors and their grouped employer affiliations

Reviewer: I don't think "substantial" contributors is part of the governance criteria, so this should probably be maintainers?

Author: Updated

    * some general information about its leadership
    * the project's general status and maturity
* If the project has any unusual aspects to its governance, describe them here.
* Link to the project's existing documents where applicable.

> **_NOTE:_** If the project is already under CNCF, you can use Devstats to check the number of substantial contributors.

##### Discoverability

###### Governance Location

* Describe where the governance documents are located. Is it in a primary repo, community Repo, somewhere else?

###### Governance Discovery Completeness

* Describe how easy is it for potential contributors to find and read the governance documentation.

> * Is it findable from the project web page?

Reviewer: ```suggestion
* Is it findable from the project web page and repository README files?
```

and contributors to monitor the project's adherence to their stated governance.

Freshness indicates governance activities mirror the documented governance for the project, and have been reviewed or updated recently.

Answer the following questions in a paragraph:
* Are governance activities transparent and monitorable?
* Are the governance documents up to date?
* Do they accurately reflect current project participants, code and subproject status, etc?

###### Governance Drift

Governance Drift can occur when the executed and observable governance of a project deviates from the documented governance of the project.

Answer the following questions in a paragraph:
* Are the governance activities being carried out?
* Are community meetings (if any) happening?
* Are required elections and votes taking place?
* Are official communications channels accessible, staffed and responsive?
* Are they being used?
* Are questions and proposed updates/changes to governance (if any) being transparently discussed and addressed?

###### Ownership

Not applicable for projects joining the CNCF.

For any CNCF projects:

> * Request that CNCF staff carry out an audit (via Sheriff) that the explicit governance of the

Reviewer: We aren't using Sheriff any longer, and I think that CLOWarden doesn't have this functionality implement yet, so might be better to leave this out and add it back in when we're ready to be able to do this.

In the meantime, I think we ask the project to do the checks below manually.

Author: Updated
